app/ante/evm/claim_rewards.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022 Evmos Foundation
+// This file is part of the Evmos Network packages.
+//
+// Evmos is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Evmos packages are distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Evmos packages. If not, see https://github.com/evmos/evmos/blob/main/LICENSE
+
+package evm
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// ClaimStakingRewardsIfNecessary checks if the given address has enough balance to cover the
+// given amount. If not, it attempts to claim enough staking rewards to cover the amount.
+func ClaimStakingRewardsIfNecessary(
+	ctx sdk.Context,
+	bankKeeper BankKeeper,
+	distributionKeeper DistributionKeeper,
+	stakingKeeper StakingKeeper,
+	addr sdk.AccAddress,
+	amount sdk.Coins,
+) error {
+	stakingDenom := stakingKeeper.BondDenom(ctx)
+	found, amountInStakingDenom := amount.Find(stakingDenom)
+	if !found {
+		// TODO: Should we return an error here? If the fees are not in the staking denom?
+		return nil
+	}
+
+	// TODO: Is this only giving the spendable balance or also the locked balance?
+	balance := bankKeeper.GetBalance(ctx, addr, stakingDenom)
+	if balance.IsNegative() {
```

Comment:
```suggestion
	if !balance.IsPositive() {
```

Reply:
@fedekunze this would raise an error if the balance is zero, which is the edge case that we want to support? Only `IsNegative` would work in that case